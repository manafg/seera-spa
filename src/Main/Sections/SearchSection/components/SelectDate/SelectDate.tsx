import { Flex, Spacer, Box, Button } from "@chakra-ui/react";
import React, { useState } from "react";
import { SearchIcon } from "@chakra-ui/icons";

import DatePickerComp from "../../../../Components/DatePicker";
import useDealConfig from "../../../../hooks/useDealConfig";

function SelectDate() {
  const { searchByDateDealMap } = useDealConfig();
  const [fromDate, setFromDate] = useState(new Date());
  const [toDate, setToDate] = useState(new Date());

  const changeFromDate = (date: Date) => {
    setFromDate(date);
  };
  const changeToDate = (date: Date) => {
    setToDate(date);
  };

  return (
    <Box>
      <Flex>
        <Box w='150px' h='30'>
          <DatePickerComp onChange={changeFromDate} selectedDatePicker={fromDate} />
        </Box>
        <Spacer />
        <Box w='150px' h='30'>
          <DatePickerComp onChange={changeToDate} selectedDatePicker={toDate} />
        </Box>
        <Spacer />
        <Box w='150px' h='30'>
          <Button leftIcon={<SearchIcon />}  onClick={()=>searchByDateDealMap(fromDate, toDate )} colorScheme='red'>
            Search
          </Button>
        </Box>
      </Flex>
    </Box>
  );
}

export default SelectDate;
