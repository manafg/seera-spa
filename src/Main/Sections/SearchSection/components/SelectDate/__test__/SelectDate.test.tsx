import { render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";

import SelectDate from "../SelectDate";

import "@testing-library/jest-dom/extend-expect";
import rrender from "react-test-renderer";

describe("Snapshot tests", () => {

    test("Matches existing Snapshot", () => {
        // @ts-ignore
        const tree = rrender.create(<SelectDate /> ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
