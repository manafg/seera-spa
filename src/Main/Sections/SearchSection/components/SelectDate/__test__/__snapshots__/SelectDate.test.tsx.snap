// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Matches existing Snapshot 1`] = `
<div
  className="css-0"
>
  <div
    className="css-k008qs"
  >
    <div
      className="css-o94p7g"
    >
      <div
        className="react-datepicker-wrapper picker"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="10/04/2021"
          />
        </div>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-o94p7g"
    >
      <div
        className="react-datepicker-wrapper picker"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className=""
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="10/04/2021"
          />
        </div>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-o94p7g"
    >
      <button
        className="chakra-button css-5c2176"
        onClick={[Function]}
        type="button"
      >
        <span
          className="chakra-button__icon css-1wh2kri"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <path
              d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
              fill="currentColor"
            />
          </svg>
        </span>
        Search
      </button>
    </div>
  </div>
</div>
`;
